laws/src/main/scala/cats/laws/UnorderedFoldableLaws.scala

```diff
@@ -0,0 +1,64 @@
+package cats
+package laws
+
+import cats.implicits._
+import cats.kernel.CommutativeMonoid
+
+trait UnorderedFoldableLaws[F[_]] {
+  implicit def F: UnorderedFoldable[F]
+
+  def unorderedFoldConsistentWithUnorderedFoldMap[A: CommutativeMonoid](fa: F[A]): IsEq[A] =
+    F.unorderedFoldMap(fa)(identity) <-> F.unorderedFold(fa)
+
+
+
+  def forallConsistentWithExists[A](fa: F[A], p: A => Boolean): Boolean = {
+    if (F.forall(fa)(p)) {
+      val negationExists = F.exists(fa)(a => !(p(a)))
+
+      // if p is true for all elements, then there cannot be an element for which
+      // it does not hold.
+      !negationExists &&
+        // if p is true for all elements, then either there must be no elements
+        // or there must exist an element for which it is true.
+        (F.isEmpty(fa) || F.exists(fa)(p))
+    } else true // can't test much in this case
+  }
+
+  def existsLazy[A](fa: F[A]): Boolean = {
+    var i = 0
+    F.exists(fa){ _ =>
+      i = i + 1
+      true
+    }
+    i == (if (F.isEmpty(fa)) 0 else 1)
+  }
+
+  def forallLazy[A](fa: F[A]): Boolean = {
+    var i = 0
+    F.forall(fa){ _ =>
+      i = i + 1
+      false
+    }
+    i == (if (F.isEmpty(fa)) 0 else 1)
+  }
+
+  /**
+    * If `F[A]` is empty, forall must return true.
+    */
+  def forallEmpty[A](fa: F[A], p: A => Boolean): Boolean = {
+    !F.isEmpty(fa) || F.forall(fa)(p)
+  }
+
+  def toSetRef[A](fa: F[A]): IsEq[Set[A]] =
+    F.unorderedFoldMap(fa)(a => Set(a)) <-> F.toSet(fa)
```

Comment:
this could be
```scala
val lst1 = F.unorderedFoldMap(fa)(a => List(a))
val lst2 = F.toList(fa)
def contains[A: Eq](a: A): Boolean = lst1.exists(Eq[A].eqv(_, a))
val allCont = list2.map(contains)
val allTrue = lst.map(_ => true)
allCont <-> allTrue
```
In this way, we have a law that only needs Eq, not unusual, and the typeclass does not require any Order or Hash.
if we have an `Order[A]`. So we use an `Order` to check the law more quickly, or we could